lib/stat_tracker.rb

```diff
@@ -54,57 +54,37 @@ def lowest_scoring_home_team
     end
   end
 
-  def highest_scoring_visitor 
-    #make a hash of of total away games by team id
-    away_games_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
-      hash[game.away_team_id] += 1
-    end
-    #make a hash of total away goals by team id
-    away_goals_by_team_id = @games.each_with_object(Hash.new(0.0)) do |game, hash|
-      hash[game.away_team_id] += game.away_goals
+  def total_away_goals 
+    total_goals = @game_teams.each_with_object({}) do |game, hash|
+      hash[game.team_id] = hash[game.team_id] || [0, 0]
+      hash[game.team_id] = [game.goals + hash[game.team_id][0], hash[game.team_id][1] + 1] if game.hoa == "away"
```

Comment:
Looks great. I think it's nice, not necessarily needed, to follow a similar pattern to ensure uniformity across files and contributors.